### DISCUSSION TITLE
Subscribe to auth provider changes

### DISCUSSION THREAD

**Discussion by Jukera** [pocketbase/pocketbase#1879] [Q&A] [answered]
Hello,

I am trying to determine if it is possible to subscribe to changes in the authentication providers of a Pocketbase instance. Specifically, I would like to know if it is possible to detect when an authentication provider is toggled on or off in the Pocketbase admin dashboard.

I have tried using the subscribe() method on the auth collection of the Pocketbase instance, but it does not seem to be working. I am unsure if this is a limitation of Pocketbase or if I am using the wrong method to subscribe to changes.

Could you please let me know if it is possible to subscribe to changes in the authentication providers, and if so, what method should be used to do so? Thank you.

**Comment by ganigeorgiev** [comment, marked as answer]
The realtime events are currently triggered only on collection record change (create, update, delete).

There are no realtime events for the auth providers or any of the other settings in general.

Auth providers can be enabled/disabled only by admins so I'm not sure that I understand your use case. Could you elaborate a little more?

In any case, if you want to intercept auth provider change, you can attach to the [`OnSettingsBeforeUpdateRequest` app event hook](https://pocketbase.io/docs/event-hooks/#settings-api-hooks):

```go
app.OnSettingsBeforeUpdateRequest().Add(func(e *core.SettingsUpdateEvent) error {
    log.Println(e.OldSettings)
    log.Println(e.NewSettings) // not saved yet

    return nil
})
```

**Comment by Jukera** [comment]
Hello,

I saw that the user authentication providers are stored in the _pb_users_auth_ collection, but when I tried to subscribe to changes in that collection, I did not receive any updates, so I was confused.

I appreciate your help. Thank you!

**Reply by ganigeorgiev** [reply]
I don't understand what do you mean by _"authentication providers are stored in the pb_users_auth collection"_.

Do you mean the `users` collection records?

I thought by "Auth providers" you meant the settings available in the "Admin UI > Settings > Auth providers".

If you want to listen to changes of the `users` collection records (or any other collection), you can do something like:

```js
pb.collection("users").subscribe("*", (e) => { ... });
```

As mentioned in https://pocketbase.io/docs/api-realtime/ and in the "API Preview > Realtime", you could subscribe to a single record or to an entire collection (aka. `*`).

When you subscribe to a single record, the collection's **View API rule** will be used to determine whether the subscriber has access to receive the event message.

When you subscribe to an entire collection (`*`), the collection's **List API rule** will be used to determine whether the subscriber has access to receive the event message.

**Or in other words, if you are not recieving any change event, make sure that your client satisfies the related collection API rules.**
